src/main/kotlin/compiler/Parser.kt

```diff
@@ -227,6 +229,37 @@ class Parser(inputTokens: List<Token>) {
         return null
     }
 
+    // Parse spread assign: [var1, var2...] = list
```

Comment:
This parser is a little weird compared to most of what I have, because I need to look ahead a fair bit to see if we really have a spread-assign situation, without actually consuming the tokens.  Hence the clunky two-stage matching of confirming a spread, and then consuming/parsing it.